src/test/java/advisor/authentication/SpotifyAccessTokenFetcherTest.java

```diff
@@ -139,14 +145,14 @@ void givenValidResponse_whenRequestingAccessToken_thenCallDoneWithCorrectHeaderA
         // THEN
         String expectedBase64EncodedClientData =
                 Base64.getEncoder().encodeToString(String.join(":", CLIENT_ID, CLIENT_SECRET).getBytes());
-        String expectedRequestBody = String.join("&",
-                List.of("code=" + accessCode,
-                        "grant_type=authorization_code",
-                        "redirect_uri=" + REDIRECT_URI));
         verify(postRequestedFor(urlEqualTo(API_TOKEN_URL_PATH))
                 .withHeader("Authorization", equalTo("Basic " + expectedBase64EncodedClientData))
-                .withHeader(CONTENT_TYPE, equalTo("application/x-www-form-urlencoded; charset=UTF-8"))
-                .withRequestBody(equalTo(expectedRequestBody)));
+                .withHeader(CONTENT_TYPE, equalTo("application/x-www-form-urlencoded; charset=UTF-8")));
+        List<LoggedRequest> requests = findAll(postRequestedFor(urlEqualTo(API_TOKEN_URL_PATH)));
+        String actualRequestBody = requests.get(0).getBodyAsString();
+        Assertions.assertTrue(actualRequestBody.contains("code=" + accessCode) &&
+                actualRequestBody.contains("grant_type=authorization_code") &&
+                actualRequestBody.contains("redirect_uri=" + REDIRECT_URI));
```

Comment:
Nice one as well. Small suggestion to improve the assertion (can be ignored)

```suggestion
        assertThat(actualRequestBody)
                .contains("code=" + accessCode)
                .contains("grant_type=authorization_code")
                .contains("redirect_uri=" + REDIRECT_URI);
```